src/ajax.js

```diff
@@ -444,80 +444,80 @@ jQuery.extend( {
 			// Default abort message
 			strAbort = "canceled",
 
-			// Fake xhr
-			jqXHR = {
-				readyState: 0,
+			// Fake xhr (ref the true Deferred instance above)
+			jqXHR = deferred;
```

Comment:
I'm sorry I missed this earlier, but the reason we're returning a promise right now is that Deferred objects have private methods (`resolveWith`, `reject`, `notify`, etc.). It's not acceptable to return an object that can circumvent the normal flow (e.g., `var jqXHR = jQuery.ajax( url ); jqXHR.resolve( fakeData, fakeStatus, jqXHR )`), and I'm not sure the benefits of this ticket are worth the cost of manually deleting them.